api/apps/api/src/modules/access-control/scenarios-acl/scenario-acl.controller.ts

```diff
@@ -0,0 +1,58 @@
+import { apiGlobalPrefixes } from '@marxan-api/api.config';
+import {
+  Controller,
+  Req,
+  Param,
+  ParseUUIDPipe,
+  UseGuards,
+  ForbiddenException,
+  Query,
+  Patch,
+  HttpCode,
+  HttpStatus,
+  Body,
+} from '@nestjs/common';
+import { JwtAuthGuard } from '@marxan-api/guards/jwt-auth.guard';
+import { ScenarioAclService } from '@marxan-api/modules/access-control/scenarios-acl/scenario-acl.service';
+import { RequestWithAuthenticatedUser } from '@marxan-api/app.controller';
+import {
+  ApiBearerAuth,
+  ApiNoContentResponse,
+  ApiOperation,
+  ApiTags,
+} from '@nestjs/swagger';
+import { isLeft } from 'fp-ts/lib/These';
+import {
+  UserRoleInScenarioDto,
+} from '@marxan-api/modules/access-control/scenarios-acl/dto/user-role-scenario.dto';
+
+@UseGuards(JwtAuthGuard)
+@ApiBearerAuth()
+@ApiTags('Scenarios-Users Roles')
+@Controller(`${apiGlobalPrefixes.v1}/roles/scenarios`)
+export class ScenarioAclController {
+  constructor(private readonly scenarioAclService: ScenarioAclService) {}
+
+  @Patch(':scenarioId/users')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ApiOperation({ summary: 'Add user and proper role to a project' })
+  @ApiNoContentResponse({
+    status: 204,
+    description: 'User was updated correctly',
+  })
+  async updateUserInProject(
+    @Body() dto: UserRoleInScenarioDto,
+    @Param('scenarioId', ParseUUIDPipe) scenarioId: string,
+    @Req() req: RequestWithAuthenticatedUser,
+  ): Promise<void | boolean> {
+    const result = await this.scenarioAclService.updateUserInProject(
+      scenarioId,
+      dto,
+      req.user.id,
+    );
```

Comment:
it isn't fully clear to me - dto hides some things (also why would it be used by service, as it is REST thing?) and name states `in project` but there is `scenarioId` given.

```
this.acl.grant(scenarioId, role, userId)
```

Reply:
We need both loggedUserId (to check ownership) and userId + roleName on the other side to change the given user into the role selected, that's why I send the dto directly to the service. Wouldn't it be more confusing if I got the things out of the DTO?

Reply:
Maybe changing the name of it into something like `UserAndRoleToChange` or whatever will be better option here.